### DISCUSSION TITLE
NMR spectra processing and analysis tools / Output formats

### DISCUSSION THREAD

**Discussion by NRayya** [NFDI4Chem/nmrxiv#296] [General]
Tools such as [ACD Labs](https://www.acdlabs.com/solutions/nmr-spectroscopy/) and [Mnova](https://mestrelab.com/software/mnova/) are widely used among the NMR research community to process and analyse NMR spectra. 

It is highly impossible and not desirable to be able to reproduce everything that these tools offer on the Chemotion and Repositories. At the same time, we cannot request the researcher to recreate the analysis using the tools we support in Chemotion/Repositories.

The output from these tools is usually stored in proprietary formats, such as .spectrus and .mnova. These formats are closed and are often updated with breaking changes etc.

To avoid losing all the rich (meta-)data from these third-party tools, it would be great if we could export (using macros - links below) the data from these tools into standard formats (nmreData, Jcamp, nmrML) rather than being able to parse their proprietary output formats. 

These tools already support export in some of the standard/widely known formats, such as Jcamp (ACD Labs) and NMReData (Mnova), which contain a subset (molecular structure, assignments, peaks, integrals etc.) of the complete data but are still valuable.

It would be great if there is a collective effort to push the ACDLabs / Mnova to enable better or uptodate export functions (or develop macros that does this conversion) and also be able to develop parsers to process the standard output file format formats with the rich meta data.

https://resources.mestrelab.com/how-to-get-started-with-mnova-scripts/
https://www.acdlabs.com/resource/automating-processes-using-macros-in-acd-spectrus-software/

Example output:
Mnova - nmreData - https://drive.google.com/file/d/1a6na29Gc8FHDP38QC19BLsOzWtj5H2sp/view?usp=sharing
ACD Labs - Jdx - https://drive.google.com/drive/folders/1P5NAVrI0tiIShLkhnckr2vHHMxMKnrzt?usp=sharing

**Comment by NRayya** [comment]
@sneumann @lpatiny We would appreciate your input

**Comment by vcnainala** [comment]
Related issues:

https://github.com/cheminfo/nmrium/issues/1735
https://github.com/cheminfo/nmrium/issues/1758

**Comment by tilfischer** [comment]
More related issues:

https://github.com/cheminfo/nmrium/issues/2005
[nmrXiv/nmrium#1758](https://github.com/NFDI4Chem/nmrxiv/issues/981)

**Comment by tilfischer** [comment]
This discussion has overarching importance to everything what is done for nmrXiv.

> It is highly impossible and not desirable to be able to reproduce everything that these tools offer on the Chemotion and Repositories. At the same time, we cannot request the researcher to recreate the analysis using the tools we support in Chemotion/Repositories.

I totally agree. Moreover, users need to know what they need to do. The best example is the NMR data of Linderazuelene, Pulegone and Carminic acid, which got published now. The data in Bruker format was obtained with TopSpin 3.2, which haven't had a feature to autoprocess FIDs, hence, there is not necessarily a pdata folder with a PROCNO containing the processed spectra (1r/1i) and processing parameters (proc, procs). I first had to find out that the processed spectra 1r/1i are required. The solution is "simple": Install TopSpin 4.3 with academic license, open the data and close TopSpin again. TopSpin 4.3 (possibly since TopSpin 3.6) has autoprocessing enabled by default. For this reason it creates the PROCNO with 1r/1i during autoprocessing, if the data is opened with this newer version of TopSpin. Then, the Bruker data works with NMRium.

It will get even more complicated when we approach Varian NMR data obtained with VnmrJ. Varian closed down in 2014. Is it legacy data we do not need to take care of? Certainly not! There are a lot of Varian spectrometers out there which ware just 10-12 years old. This spectrometers will continue to produce data for at least the next 10 years. 

> The output from these tools is usually stored in proprietary formats, such as .spectrus and .mnova. These formats are closed and are often updated with breaking changes etc.
> 
> To avoid losing all the rich (meta-)data from these third-party tools, it would be great if we could export (using macros - links below) the data from these tools into standard formats (nmreData, Jcamp, nmrML) rather than being able to parse their proprietary output formats.
> 
> These tools already support export in some of the standard/widely known formats, such as Jcamp (ACD Labs) and NMReData (Mnova), which contain a subset (molecular structure, assignments, peaks, integrals etc.) of the complete data but are still valuable.

Commercial companies will certainly fix their export as JCAMP-DX as soon as users who pay for that software are complaining. However, they will state that they must make additions to the JCAMP-DX files they export by excessively using comments ``$$`` and private labels ``##$`` and generate their own JCAMP-DX flavour or dialect as the JCAMP-DX file would otherwise not be able to hold the required data. What is required is a well described and defined open data format having reference readers and writer as well as a reference validator. 

Best,
Tillmann
